### DISCUSSION TITLE
Error connecting SQL Server 2000

### DISCUSSION THREAD

**Discussion by ClaudioNoronha** [Budibase/budibase#7406] [General]
Hello. Has anyone here used Budibase with Microsoft SQL Server 2000? I configured the connection to the database, but when I try to fetch the tables, I get a connection error message. I've already tested the connection using python code and also through SQL Server Management Studio and it connects normally.

The error message I get is:

Error updating datasource schema : ConnectionError: Failed to connect to sbs-sql-server:1433 - getaddrinfo ENOTFOUND sbs-sql-server

![image](https://user-images.githubusercontent.com/75019513/185964162-61b3a389-5c8e-497b-94e2-d139f951e500.png)

**Comment by andz-bb** [comment]
Hi Claudio, from reading #7273 I believe that MS SQL Server 2000 should be supported by Budibase but I don't have an instance to test it on my end. Have you had the chance to try any more recent versions of SQL Server with Budibase? When you try to fetch the tables do you get a more detailed error message in your browser's console or network tab in developer tools? Thanks

**Reply by ClaudioNoronha** [reply]
Hello @andz-bb . Thanks for your answer

I also tried with SQL Server 2019 and the same error occurs. Below the console logs.

-----------------------------------------------------------------------------------------
index.d953a9aa.js:377          POST http://localhost:10000/api/datasources/datasource_plus_9327deda28344e71888fd27a7bad9518/schema 500 (Internal Server Error)
l @ index.d953a9aa.js:377
(anonymous) @ index.d953a9aa.js:377
buildDatasourceSchema @ index.d953a9aa.js:377
updateSchema @ index.d953a9aa.js:377
T @ index.d953a9aa.js:406
G @ index.d953a9aa.js:23
(anonymous) @ vendor.6bcafa60.js:4
bQ @ vendor.6bcafa60.js:4
T @ index.d953a9aa.js:14
(anonymous) @ vendor.6bcafa60.js:1
index.d953a9aa.js:1 [Builder] HTTP 500 on POST:http://localhost:10000/api/datasources/datasource_plus_9327deda28344e71888fd27a7bad9518/schema
	ConnectionError: Failed to connect to growz:1433 - getaddrinfo ENOTFOUND growz
TE.console.warn @ index.d953a9aa.js:1
onError @ index.d953a9aa.js:377
(anonymous) @ index.d953a9aa.js:377
await in (anonymous) (async)
buildDatasourceSchema @ index.d953a9aa.js:377
updateSchema @ index.d953a9aa.js:377
T @ index.d953a9aa.js:406
G @ index.d953a9aa.js:23
(anonymous) @ vendor.6bcafa60.js:4
bQ @ vendor.6bcafa60.js:4
T @ index.d953a9aa.js:14
(anonymous) @ vendor.6bcafa60.js:1
--------------------------------------------------------------------------------------------------------
Follow the screens with the errors

![image](https://user-images.githubusercontent.com/75019513/186447245-7654860f-57c0-4aaa-ac6d-e8393fa073e6.png)

![image](https://user-images.githubusercontent.com/75019513/186447605-de1a779f-ca51-4f14-b308-bc97260da631.png)

**Reply by andz-bb** [reply]
hey @claudionoronhafilho, thanks for the detailed reply! I'm thinking your value for Server maybe isn't right, it needs to be a valid IP or domain here. What are the details you use when you connect using SQL Server Management Studio?

**Reply by ClaudioNoronha** [reply]
Hi @andz-bb SQL Server is installed locally. Even using the local IP (192.168.1.3), the same error appears on the computer where it is installed. Here's a screenshot of SQL Management Studio. 

![image](https://user-images.githubusercontent.com/75019513/186487991-94a04205-af52-4ecf-9a03-def65739e531.png)

**Reply by ClaudioNoronha** [reply]
Hi @andz-bb . Good news. I was successful in connecting SQL Server 2008. I needed to enable access via TCP in SQL Server and free the port on the firewall. Now I'm able to access both locally using the internal IP, and externally using ngrok.

I will try the same with SQL Server 2000 which is the DB I need to access.
